core/classifiers/CodeClassifier/CodeClassifier.py

```diff
@@ -0,0 +1,481 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Base class for classification algorithms"""
+
+from collections import Counter
+import keyword
+from StringIO import StringIO
+import token
+import tokenize
+
+from core.classifiers.BaseClassifier import BaseClassifier
+
+import numpy as np
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
+
+# pylint: disable=invalid-name
```

Comment:
There are some variable names like `T, K, A, B` which are invalid-name. `T` and `K` are model parameters of winnowing (research paper of winnowing names them that way) so I decided to keep their name as it is. `A` and `B` are used in jaccard index function. They can be changed to something else but I'd rather not change `T` and `K`.